Integrations/integration-ATD.yml

```diff
@@ -37,672 +38,670 @@ configuration:
+        return {
+            'Accept': 'application/vnd.ve.v1.0+json',
+            'VE-SDK-API': base64.b64encode(sess['session'] + ':' + sess['userId'])
         }
-    }
 
-    function getHeaders(){
-        var sess = getSessionCredentials();
-        return {'Accept': ['application/vnd.ve.v1.0+json'],
-                'VE-SDK-API':[Base64.encode(sess.session + ':' + sess.userId)]
-        };
-    }
-
-    function sendRequestRaw(uri, method, headers, body){
-        var res = http(
-                BASE+uri,
-                {
-                    Method: method,
-                    Headers: headers,
-                    Body: body
-                },
-                UNSECURE,
-                PROXY
-            );
-        if (res.StatusCode < 200 || res.StatusCode >= 300) {
-            throw 'Failed to use uri ' + BASE+uri + ', request status code: ' + res.StatusCode + ' and Body: ' + res.Body + '.';
-        }
-        var uriStr = String(uri);
-        if (uriStr.indexOf("session") === -1) {
-            logout();
+    def http_request(uri, method, headers = {}, body = {}, params={}, files={}):
+        ''' Makes an API call with the supplied uri, method, headers, body '''
+        url = '%s/%s' % (BASE_URL, uri)
+        res = requests.request(
+            method,
+            url,
+            headers=headers,
+            data=body,
+            verify=USE_SSL,
+            params=params,
+            files=files
+        )
+        if res.status_code < 200 or res.status_code >= 300:
+                return_error('Got status code ' + str(res.status_code) + ' with body ' + res.content + ' with headers ' + str(res.headers))
+        result = res.content
+        if not uri.startswith('php/showreport.php?'):
+            result = json.loads(result)
+            if 'success' in result:
+                if (result['success'] == 'false'):
+                    return_error('ATD Api call to ' + uri + ' failed with request body: ' + body)
+        return result
+
+
+
+    ''' FUNCTIONS '''
+    def heart_beat():
+        return http_request('php/heartbeat.php','get', API_HEADERS, HEARTBEATHEADERS)
+
+    def test_get_session():
+        res = http_request('php/session.php','get', LOGINHEADERS)
+
+    def prettify_current_user_res(currentUser):
+        prettyCurrentUser = {
+            'APIVersion': currentUser['apiVersion'],
+            'IsAdmin': 'True' if (currentUser['isAdmin'] == '1') else 'False',
+            'SessionId': currentUser['session'],
+            'UserId': currentUser['userId']
         }
-        return res;
-    }
-
-    function sendRequest(uri, method, headers, body){
-        body = sendRequestRaw(uri,method, headers, body).Body;
-        var res = JSON.parse(body);
-        if (res.success) {
-            if (res.success === 'false') {
-                throw 'ATD Api call to '+uri+' failed with request body: '+body;
+        return prettyCurrentUser
+
+    def get_session():
```

Comment:
Shouldn't be in helper functions section?

Reply:
done